src/debuginfo.cpp

```diff
@@ -1139,29 +1156,31 @@ bool jl_dylib_DI_for_fptr(size_t pointer, const llvm::object::ObjectFile **obj,
             debug_link_info info = getDebuglink(*debugobj);
             if (!info.filename.empty()) {
                 size_t sep = fname.rfind('/');
-                ErrorOr<object::OwningBinary<object::ObjectFile>> DebugInfo(std::errc::no_such_file_or_directory);
+#ifdef LLVM39
+                Expected<object::OwningBinary<object::ObjectFile>>
+                    DebugInfo(errorCodeToError(std::make_error_code(std::errc::no_such_file_or_directory)));
+#else
+                ErrorOr<object::OwningBinary<object::ObjectFile>>
+                    DebugInfo(std::errc::no_such_file_or_directory);
+#endif
                 if (fname.substr(sep + 1) != info.filename) {
                     debuginfopath = fname.substr(0, sep + 1);
                     debuginfopath += info.filename;
                     DebugInfo = openDebugInfo(debuginfopath, info);
                 }
-                if (DebugInfo.getError()) {
+                if (!DebugInfo) {
```

Comment:
`takeError` *
Yeah, I guess I also need to check how moving assignments are handled and if the error needs to be explicitly consumed there too...... (potentially many more `#ifdef LLVM39` coming......)